TST: add ASV check for groupby with uint64 col

smithto1 opened the pull request:
- [x] closes #28122 
- [x] tests added/passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

The performance regression reported in the original issue was fixed somewhere between 0.25.1 and 1.2.4. This PR just adds an ASV test to monitor for the same regression in the future.

smithto1 committed:
GH #28122 add ASV check for groupby with uint64 col

WillAyd commented:
Thanks for the PR - can you post the result to be sure? Also do you think we could just parametrize `GroupByMethods` to include this instead? That may be slightly more comprehensive than this one-off class, though noted that you used a similar class for Int as inspiration

smithto1 committed:
moving uint ASV into GroupByMethods, removed standalone UInt64 class

smithto1 commented:
> Thanks for the PR - can you post the result to be sure? Also do you think we could just parametrize `GroupByMethods` to include this instead? That may be slightly more comprehensive than this one-off class, though noted that you used a similar class for Int as inspiration

@WillAyd I prefer your suggestion to run `uint` through `GroupByMethods`. I've implemented this way and removed the one-off class. 

Results attached. 

[asv_groupby.GroupByMethods.txt.txt](https://github.com/pandas-dev/pandas/files/6617853/asv_groupby.GroupByMethods.txt.txt)
